### ISSUE TITLE
mediafile.upload

### ISSUE THREAD

**Issue by FinnStutzenstein** [OpenSlides/openslides-backend-coordination#20]
~~TODO: Should we split it up into `mediafile.upload` and `mediafile.create_directory`?~~

Done with #137

**Comment by reiterl** [comment]
Opened https://github.com/OpenSlides/openslides-backend/pull/274 for this.

**Comment by jsangmeister** [comment]
merged.

**Comment by FinnStutzenstein** [comment]
- `is_directory` must be added to the data, so the client can distinguish between files and directories.
- `filesize`, `mimetype`, `create_timestamp` are missing
- `has_inherited_access_groups` has to be renamed to `is_public`

Possibly `pdf_information` is also missing, not tested.

**Comment by FinnStutzenstein** [comment]
https://github.com/OpenSlides/openslides-backend/pull/297 to fix some small issues.

**Comment by jsangmeister** [comment]
> * `is_directory` must be added to the data, so the client can distinguish between files and directories.

Shouldn't it be enough that it is not set and therefore `null == false`?

> * `has_inherited_access_groups` has to be renamed to `is_public`

That's my bad, that comes from handling multiple PRs in parallel. Will fix

> Possibly `pdf_information` is also missing, not tested.

What has to be written to `pdf_information`?

**Comment by FinnStutzenstein** [comment]
> Shouldn't it be enough that it is not set and therefore `null == false`?

I've listed it since the presenter crashed (KeyError on `mediafile["is_directory"]`))...

> What has to be written to pdf_information?

I guess, that the `get_pdf_information` can be copied to the backend code 1:1 (attention, new dependency):
https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/mediafiles/utils.py

This is only done, if the mimetype is `application/pdf`

**Comment by jsangmeister** [comment]
Should be finished now. Please check again.
